tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -1457,9 +1457,50 @@ TEST_F(EnergyPlusFixture, ScheduleFileDSTtoggleOptionTest)
     EXPECT_DOUBLE_EQ(ScheduleManager::LookUpScheduleValue(*state, sch4idx, state->dataGlobal->HourOfDay, state->dataGlobal->TimeStep), 1.0);
 }
 
-TEST_F(EnergyPlusFixture, ShadowCalculation_CSV_extra_parenthesis)
+TEST_F(EnergyPlusFixture, ScheduleFile_Blanks)
 {
 
+    // On the third line (second data record after header), there is a blank in the second column
+    fs::path scheduleFile = configured_source_directory() / "tst/EnergyPlus/unit/Resources/schedule_file_with_blank.csv";
+
+    std::string const idf_objects = delimited_string({
+        "Schedule:File,",
+        "  Test1,                   !- Name",
+        "  ,                        !- Schedule Type Limits Name",
+        "  " + scheduleFile.string() + ",              !- File Name",
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60,                      !- Minutes per item",
+        "  Yes;                     !- Adjust Schedule for Daylight Savings",
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->dataGlobal->NumOfTimeStepInHour = 4; // must initialize this to get schedules initialized
+    state->dataGlobal->MinutesPerTimeStep = 15; // must initialize this to get schedules initialized
+    state->dataGlobal->TimeStepZone = 0.25;
+    state->dataGlobal->TimeStepZoneSec = state->dataGlobal->TimeStepZone * Constant::SecInHour;
+    state->dataEnvrn->CurrentYearIsLeapYear = false;
+
+    ASSERT_THROW(ScheduleManager::ProcessScheduleInput(*state), EnergyPlus::FatalError); // read schedules
+
+    const std::string expected_error = delimited_string({
+        "   ** Severe  ** CsvParser - Line 3 - Expected 3 columns, got 2. Error in following line.",
+        "   **   ~~~   ** 1,,0.33",
+        "   **   ~~~   ** Error Occurred in " + scheduleFile.string(),
+        "   **  Fatal  ** Program terminates due to previous condition.",
+        "   ...Summary of Errors that led to program termination:",
+        "   ..... Reference severe error count=1",
+        "   ..... Last severe error=CsvParser - Line 3 - Expected 3 columns, got 2. Error in following line.",
+    });
+
+    compare_err_stream(expected_error);
```

Comment:
Ensure we get a fatal with useful diagnostics

Reply:
Does this need to be fatal? Could we assume zero for blank and just throw a warning? I can see if it was missing the comma, `1,0.33` then that should be fatal.

Reply:
@Myoldmopar suggested throwing when I asked about it, and I tend to agree.

I don't think 0 is really a safe default. But if people agree it is, happy to make the change.

Reply:
If you and @Myoldmopar are in agreement, then it's fine with me.